lib/foreman_theme_satellite/documentation.rb

```diff
@@ -77,28 +77,38 @@ module Documentation
       'foreman_discovery' => "#{ForemanThemeSatellite.documentation_root}/provisioning_hosts/discovering-hosts-on-a-network_provisioning",
     }.freeze
 
-    DOCS_GUIDES_LINKS = {
-      'Managing_Content' => {
-        'Products_and_Repositories_content-management' => "#{ForemanThemeSatellite.documentation_root}/managing_content/importing_content_content-management#Products_and_Repositories_content-management",
-      },
+    # Guide mapping. If no entry is present, it should be assumed that it can
+    # be downcased
+    DOCS_GUIDE_MAPPING = {
+      'Managing_Configurations_Ansible' => 'managing_configurations_using_ansible_integration',
+    }.freeze
+
+    # An upstream chapter mapping, in case downstream another chapter should be
+    # used. The top level key is the upstream guide name where the value is a
+    # mapping of upstream chapter to downstream mapping
+    DOCS_GUIDE_CHAPTER_MAPPING = {
       'Managing_Hosts' => {
-        'registering-a-host_managing-hosts' => "#{ForemanThemeSatellite.documentation_root}/managing_hosts/registering_hosts_to_server_managing-hosts#Registering_Hosts_by_Using_Global_Registration_managing-hosts",
-      }
+        'registering-a-host_managing-hosts' => 'Registering_Hosts_by_Using_Global_Registration_managing-hosts',
+      },
     }.freeze
 
-    def self.flat_docs_guides_links
-      nested_to_flat_k_v(nil, DOCS_GUIDES_LINKS).to_h
-    end
+    DOCS_GUIDE_PAGES = JSON.load_file(File.join(__dir__, 'documentation-toc.json'))
+
+    def self.docs_url(guide, chapter)
+      root = ForemanThemeSatellite.documentation_root
+      downstream_guide = DOCS_GUIDE_MAPPING.fetch(guide) { guide.downcase }
+
+      if chapter
+        downstream_chapter = DOCS_GUIDE_CHAPTER_MAPPING.fetch(guide, chapter) || chapter
 
-    private_class_method def self.nested_to_flat_k_v(prefix, source)
-      key_values = []
-      source.map do |k, v|
-        key = "#{prefix}/#{k}"
-        if v.is_a?(Hash)
-          key_values.append(*nested_to_flat_k_v(key, v))
+        if (page = DOCS_GUIDE_PAGES[downstream_guide]&.find { |_page, chapters| chapters.include?(downstream_chapter) }&.first)
+          "#{root}/#{downstream_guide}/#{page}##{downstream_chapter}"
         else
-          key_values.append([key, v])
+          Rails.logger.warn("Could not find page for chapter '#{downstream_chapter}' in guide '#{downstream_guide}'")
```

Comment:
The rake task fails because it hasn't initialized Rails so `Rails.logger` is `nil`. I wonder what to do. It may work to use
```suggestion
          Rails.logger&.warn("Could not find page for chapter '#{downstream_chapter}' in guide '#{downstream_guide}'")
```

Another approach is to move `Rails.logger` to the caller.

And we can make the validation initialize Rails.

Thoughts?

Reply:
The user can do nothing about this, I think we shouldn't log it as warning. It's either info or not logging it at all. 

Specifically for the tests, let's inject the logger: 
``` ruby
def self.docs_url(guide, chapter, logger:  Foreman::Logging.logger('app'))
```
So for regular work it will take the logger from the rails app (specifically Foreman logger), and for the rake task we will inject a simple ruby logger :
``` ruby
require 'logger'
docs_url('foo', 'bar', logger:  Logger.new(STDOUT))
```
I do something similar in [rh_cloud](https://github.com/theforeman/foreman_rh_cloud/blob/e939a23815de0acb3e9c321321caa7688f69f641/lib/tasks/insights.rake#L26)

Reply:
Implemented